src/Features/LanguageServer/Protocol/ILanguageServerFactory.cs

```diff
@@ -15,6 +16,7 @@ internal interface ILanguageServerFactory
             JsonRpc jsonRpc,
             ICapabilitiesProvider capabilitiesProvider,
             WellKnownLspServerKinds serverKind,
-            ILspServiceLogger logger);
+            ILspServiceLogger logger,
+            HostServices hostServices);
```

Comment:
to avoid having MS.CA.LanguageServer.Protocol depend on vs-mef, we use HostServices instead of ExportProvider.  Then we import the ExportProvider at the layers that are already using it (e.g. EditorFeatures)